libraries/common/array.effekt

```diff
@@ -81,7 +82,14 @@ extern global def unsafeSet[T](arr: Array[T], index: Int, value: T): Unit =
   """
   vm "array::unsafeSet[T](Array[T], Int, T)"
 
-
+/// Fills a given array in-place.
+extern def fill[T](arr: Array[T], filler: T): Unit = 
+  js "${arr}.fill(${filler})"
+  chez "(vector-fill! ${arr} ${filler})"
+  llvm """
+    %z = call %Pos @c_array_fill(%Pos ${arr}, %Pos ${filler})
+    ret %Pos %z
+  """
```

Comment:
Nice!
We should be able to use this function in the `array::array` constructor now, right?

Reply:
It would even be nice to look at other uses of the `each(0, arr.size) { i => arr.unsafeSet(i, x) }` idiom in the codebase, I think there are many places in `io` where we use that idiom to fill an array with a value `x` (which should now be replaceable with just `arr.fill(x)`...

Reply:
Hm looking through the result of searching for the regex `each\(0,[\s]*[\w]*.size\)` shows no other occurrence of this idiom using arrays. There are some occurrences where this is done for `bytearray`s.